File: backend/src/main/java/com/punarmilan/service/AdminUserService.java
package com.punarmilan.service;

import com.punarmilan.dto.AdminUserDetailDTO;
import com.punarmilan.dto.AdminUserSearchCriteria;
import com.punarmilan.entity.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface AdminUserService {
    Page<User> getAllUsers(AdminUserSearchCriteria criteria, Pageable pageable);

    AdminUserDetailDTO getUserById(Long userId);

    void blockUser(Long userId);

    void unblockUser(Long userId);

    void deleteUser(Long userId);
}

File: punarmilan-frontend/src/admin/services/adminUserService.js
import adminApi from './adminApi';

const adminUserService = {
    getAllUsers: async (params) => {
        // params will include page, size, and filters (email, mobileNumber, gender, etc.)
        const response = await adminApi.get('/users', { params });
        return response.data;
    },

    getUserById: async (id) => {
        const response = await adminApi.get(`/users/${id}`);
        return response.data;
    },

    blockUser: async (id) => {
        const response = await adminApi.post(`/users/${id}/block`);
        return response.data;
    },

    unblockUser: async (id) => {
        const response = await adminApi.post(`/users/${id}/unblock`);
        return response.data;
    },

    deleteUser: async (id) => {
        const response = await adminApi.delete(`/users/${id}`);
        return response.data;
    }
};

export default adminUserService;
